File: examples/sid/sid.py
import iam_floyd as statement
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')

s = (
    # doc-start
    statement.Ec2('MYSID') \
        .allow() \
        .to_start_instances() \
        .to_stop_instances()
    # doc-end
)

all = [s]
out(all)
deploy(all)

File: examples/sid/sid.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import { Statement } from '../../lib';

function getStatement() {
  return (
    // doc-start
    new Statement.Ec2('MYSID') //
      .allow()
      .toStartInstances()
      .toStopInstances()
    // doc-end
  );
}
const s = [getStatement()];
out(s);
deploy(s);
